lib/libfrr.c

```diff
@@ -946,6 +946,20 @@ static int frr_config_read_in(struct thread *t)
 	return 0;
 }
 
+void frr_load_module(const char *name)
+{
+	const char *dir;
+	dir = di->module_path ? di->module_path : frr_moduledir;
+	char moderr[256];
+	struct frrmod_runtime *module;
+
+	module = frrmod_load(name, dir, moderr, sizeof(moderr));
```

Comment:
Why can't ldp call this?

Reply:
Today, frr_init() calls frrmod_load to load modules.

LDP consists of 3 processes: ldpd, ldpe, lde.

The main LDP process (ldpd) is an FRR client.  ldpd calls frr_init.  This means that ldpd will load modules via this frr_init/frrmod_load call path.

The other 2 LDP processes (ldpe, lde) are not FRR clients.  They do not call frr_init.  However, ldpe and lde contain the information that that LDP MIBs expose.

The purpose of this change is to allow ldpe and lde to load the LDP snmp module.

Thanks, Karen

Reply:
Yes, I understand that. I'm asking why ldep and lde cannot call `frrmod_load`; in other words, why do we need a new function in libfrr? I see `frrmod_load` is already publicly exposed via `module.h`:
```
qlyoung@corax ~/p/frr (master)> rg "frrmod_load"
lib/module.h
92:extern struct frrmod_runtime *frrmod_load(const char *spec, const char *dir,
```
I don't see what this function is adding except a bit of path formatting that could just as well be done in ldpe / lde.

Reply:
Btw, the commit message subject line should start with `lib:` not `lib/libfrr`

Reply:
That is true.  frrmod_load is a publicly exposed api.
Another approach would be to move frr_load_module() to ldpd.c.

When I coded this, I put frr_load_module() in libfrr.c because global variable frr_moduledir existed in libfrr.c. 

At the time, I figured any code involved with loading modules should reside in libfrr.c. 
If the consensus is that frr_load_module() belongs in ldpd.c, there's nothing that prevents doing this.

Thanks, Karen

p.s. I changed to commit subject line from lib/libfrr to lib.

Reply:
> When I coded this, I put frr_load_module() in libfrr.c because global variable frr_moduledir existed in libfrr.c.

`frr_moduledir` is also exposed in `libfrr.h`, so scoping isn't a reason this function needs to be here either.

> At the time, I figured any code involved with loading modules should reside in libfrr.c.

I kinda see the utility of a wrapper that takes just the module name and handles the path stuff for you, but if anything it probably belongs in `module.c`. Though it shouldn't call `exit(1)`, it should return an error code or something.

Reply:
Thanks, I did not realize that frr_moduledir is exposed in a header file.

I will go ahead and move this frr_load_module API to LDP.  

Thanks for your review comments.